src/Pipeline.h

```diff
@@ -391,41 +391,42 @@ class Pipeline {
 
 };
 
-namespace {
+struct ScalarOrBufferT {
```

Comment:
Type can represent buffer_t pointer args (via type_of<struct buffer_t *>()), so I don't think this class is necessary anymore.